std/machines/small_field/rotate.asm

```diff
@@ -0,0 +1,71 @@
+use std::utils::unchanged_until;
+use std::utils::cross_product;
+use std::convert::int;
+
+// Rotate for single bytes using an exhaustive table, returning two 16-bit values as result.
+// TODO this way, we cannot prove anything that rotates by more than 31 bits.
+machine ByteRotate with
+    latch: latch,
+    operation_id: operation_id,
+    degree: 65536
+{
+    // P_C0 and P_C1 are both 16 bit limbs of P_C, where P_C0 is the less significant limb.
+    operation run<0> P_operation, P_A, P_B, P_ROW -> P_C0, P_C1;
+
+    col fixed latch = [1]*;
+    col fixed operation_id = [0]*;
+
+    let bit_counts = [256, 32, 4, 2];
+    let min_degree = std::array::product(bit_counts);
+    std::check::assert(std::prover::min_degree() >= std::array::product(bit_counts), || "The rotate machine needs at least 65536 rows to work.");
+    std::check::assert(std::field::modulus() >= 65535, || "The field modulo should be at least 2^16 - 1 to work in the rotate machine.");
```

Comment:
```suggestion
    std::check::assert(std::field::modulus() >= 2**16, || "The field modulo should be at least 2^16 - 1 to work in the rotate machine.");
```

Reply:
Or, this `require_field_bits(16, || "The field modulo should be at least 2^16 - 1 to work in the shift16 machine.");`?

Reply:
Or, this? `require_field_bits(16, || "The field modulo should be at least 2^16 - 1 to work in the rotate machine.");` Which one? It's better to have similar concept with the shift machine, which uses https://github.com/powdr-labs/powdr/blob/main/std/machines/small_field/shift.asm#L24

Reply:
I think using the helper function is more descriptive.